impl_core/src/test/java/io/opencensus/implcore/tags/propagation/CorrelationContextFormatTest.java

```diff
@@ -0,0 +1,255 @@
+import io.opencensus.tags.propagation.TagContextTextFormat;
+import io.opencensus.tags.propagation.TagContextTextFormat.Getter;
+import io.opencensus.tags.propagation.TagContextTextFormat.Setter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Random;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link CorrelationContextFormat}. */
+@RunWith(JUnit4.class)
+public class CorrelationContextFormatTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  private final TagsComponent tagsComponent = new TagsComponentImplBase();
+  private final TagContextTextFormat textFormat =
+      tagsComponent.getTagPropagationComponent().getCorrelationContextFormat();
+
+  private static final TagMetadata METADATA_NO_PROPAGATION =
+      TagMetadata.create(TagTtl.NO_PROPAGATION);
+
+  private static final Setter<Map<String, String>> setter =
+      new Setter<Map<String, String>>() {
+        @Override
+        public void put(Map<String, String> carrier, String key, String value) {
+          carrier.put(key, value);
+        }
+      };
+
+  private static final Getter<Map<String, String>> getter =
+      new Getter<Map<String, String>>() {
+        @Nullable
+        @Override
+        public String get(Map<String, String> carrier, String key) {
+          return carrier.get(key);
+        }
+      };
+
+  private static final TagKey K1 = TagKey.create("k1");
+  private static final TagKey K2 = TagKey.create("k2");
+  private static final TagValue V1 = TagValue.create("v1");
+  private static final TagValue V2 = TagValue.create("v2");
+  private static final Tag T1 = Tag.create(K1, V1, METADATA_UNLIMITED_PROPAGATION);
+  private static final Tag T2 = Tag.create(K2, V2, METADATA_UNLIMITED_PROPAGATION);
+
+  private static final Random random = new Random();
+
+  @Test
+  public void fieldsList() {
+    assertThat(textFormat.fields()).containsExactly(CORRELATION_CONTEXT);
+  }
+
+  @Test
+  public void headerNames() {
+    assertThat(CORRELATION_CONTEXT).isEqualTo("Correlation-Context");
+  }
+
+  @Test
+  public void inject() throws TagContextSerializationException {
+    Map<String, String> carrier = new HashMap<String, String>();
+    textFormat.inject(makeTagContext(T1, T2), carrier, setter);
+    assertThat(carrier).containsExactly(CORRELATION_CONTEXT, "k1=v1,k2=v2");
+  }
+
+  @Test
+  public void inject_Empty() throws TagContextSerializationException {
+    Map<String, String> carrier = new HashMap<String, String>();
+    textFormat.inject(makeTagContext(), carrier, setter);
+    assertThat(carrier).containsExactly(CORRELATION_CONTEXT, "");
+  }
+
+  @Test
+  public void inject_SkipNonPropagatingTag() throws TagContextSerializationException {
```

Comment:
May be worth adding a test that intermixes propagating and non-propagating tags.

Reply:
SG, added.